### PR TITLE
Declare errno.h

### PR DESCRIPTION
**Pull request by NeKroFR**

### CODE REVIEW

**Comment by Keyboard-Slayer** [comment]
Your variables has no values ??

**Comment by Keyboard-Slayer** [comment]
And why having variable on a header file ?

**Comment by NeKroFR** [comment]
@Keyboard-Slayer I just folow the[ ISO C](http://www.open-std.org/jtc1/sc22/wg14/www/docs/n2478.pdf) doc  at the chapter

**7.5 Errors <errno.h>**

Maybe I don't enderstand something 🤔 but i think it will be weird if the headerfile was like this:

```

#pragma once
/* --- 7.5 Errors <errno.h> ----------------------------------------------- */

iEDOM
EILSEQ
ERANGE

errno
```
^^^^
This is not valid C i think

File: sources/libc/ansi/errno.h [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+/* --- 7.5 Errors <errno.h> ----------------------------------------------- */
+
+int EDOM;
```
Comment (Supercip971) [inline sources/libc/ansi/errno.h:5]: these should be a macro with a value

File: sources/libc/ansi/errno.h [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+int EILSEQ;
+
+int ERANGE;
+
+int errno;
```
Comment (Supercip971) [inline sources/libc/ansi/errno.h:11]: you should set errno to zero on entry function

**Review by Supercip971 — CHANGES_REQUESTED** [review]

File: sources/libc/ansi/errno.h (hunk 1 above)
Comment (NeKroFR) [reply sources/libc/ansi/errno.h]: I made some changes it is okay now?

File: sources/libc/ansi/errno.h (hunk 2 above)
Comment (NeKroFR) [reply sources/libc/ansi/errno.h]: I made some changes it is okay now?

File: sources/libc/ansi/errno.h [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+/* --- 7.5 Errors <errno.h> ----------------------------------------------- */
+
+#define EDOM (x);
+
+#define EILSEQ;
```
Comment (Supercip971) [inline sources/libc/ansi/errno.h:7]: they should have a value and don't use ";"

File: sources/libc/ansi/errno.h [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+#define ERANGE;
+
+int errno 0;
```
Comment (Supercip971) [inline sources/libc/ansi/errno.h:11]: maybe errno = 0; ?

**Review by Supercip971 — CHANGES_REQUESTED** [review]

File: sources/libc/ansi/errno.h [hunk 5]
```diff
@@ -4,8 +4,8 @@
 
 #define EDOM (x);
```
Comment (sleepy-monax) [inline sources/libc/ansi/errno.h:5]: You don't put ; at the end of macro declaration.

**Pull request merged**